.github/actions/on_device_tests/action.yaml

```diff
@@ -89,3 +92,22 @@ runs:
           --label author-${GITHUB_PR_HEAD_USER_LOGIN:-$GITHUB_COMMIT_AUTHOR_USERNAME} \
           --label author_id-${GITHUB_PR_HEAD_USER_ID:-$GITHUB_COMMIT_AUTHOR_EMAIL}
       shell: bash
+    - name: Download ${{ matrix.platform }} Test Results
+      if: env.TEST_TYPE == 'unit_test'
+      run: |
+        set -eux
+        gsutil -d cp "${GCS_RESULTS_PATH}" results.zip
```

Comment:
I think with 'set -e' option bash will exit if line 99 fails.  You probably need to set -e after you check gsutil return status.

Reply:
Removing that check altogether instead. It doesn't add anything useful over the regular step failed status.